src/main/java/de/tum/in/www1/artemis/service/StatisticsService.java

```diff
@@ -0,0 +1,96 @@
+     *
+     * @param span DAY,WEEK,MONTH or YEAR depending on the active tab in the view
+     * @return a array, containing the values for each bar in the graph
+     */
+    public Integer[] getTotalSubmissions(String span) {
+        switch (span) {
+            case "DAY": // result = this.statisticsRepository.getTotalSubmissionsDay(ZonedDateTime.now().minusDays(7));
+                return null;
+            case "WEEK":
+                Integer[] result = new Integer[7];
+                Arrays.fill(result, 0);
+                ZonedDateTime border = ZonedDateTime.now().minusDays(6).withHour(0).withMinute(0).withSecond(0);
+                List<Map<String, Object>> outcome = this.statisticsRepository.getTotalSubmissionsWeek(border);
+                for (Map<String, Object> map : outcome) {
+                    ZonedDateTime date = (ZonedDateTime) map.get("day");
+                    Integer amount = map.get("amount") != null ? ((Long) map.get("amount")).intValue() : null;
+                    for (int i = 0; i < 7; i++) {
+                        if (date.getDayOfMonth() == ZonedDateTime.now().minusDays(i).getDayOfMonth()) {
+                            result[6 - i] += amount;
+                        }
+                    }
+                }
+                return result;
+            case "MONTH":
+                return null;
+            case "YEAR":
+                return null;
+            default:
+                return null;
+        }
+    }
+
+    public Integer getReleasedExercises(Long span) {
+        return this.statisticsRepository.getReleasedExercises(ZonedDateTime.now().minusDays(span), ZonedDateTime.now());
+    }
+
+    public Integer getExerciseDeadlines(Long span) {
+        return this.statisticsRepository.getExerciseDeadlines(ZonedDateTime.now().minusDays(span), ZonedDateTime.now());
+    }
+
+    public Integer getConductedExams(Long span) {
+        return this.statisticsRepository.getConductedExams(ZonedDateTime.now().minusDays(span), ZonedDateTime.now());
+    }
+
+    public Integer getExamParticipations(Long span) {
+        return this.statisticsRepository.getExamParticipations(ZonedDateTime.now().minusDays(span));
+    }
+
+    public Integer getExamRegistrations(Long span) {
+        return this.statisticsRepository.getExamRegistrations(ZonedDateTime.now().minusDays(span), ZonedDateTime.now());
+    }
+
+    public Integer getActiveTutors(Long span) {
+        return this.statisticsRepository.getActiveTutors(ZonedDateTime.now().minusDays(span));
+    }
+
+    public Integer getCreatedResults(Long span) {
+        return this.statisticsRepository.getCreatedResults(ZonedDateTime.now().minusDays(span));
+    }
+
+    public Integer getResultFeedbacks(Long span) {
+        Integer result = this.statisticsRepository.getResultFeedbacks(ZonedDateTime.now().minusDays(span));
+        return (result != null ? result : 0);
```

Comment:
You do not need parentheses here

Reply:
[Done](https://github.com/ls1intum/Artemis/pull/2471/commits/499bb9fd70d8f4710fe3744189d0493f61a8db7d)